solr/solr-ref-guide/src/neural-search.adoc

```diff
@@ -0,0 +1,733 @@
+= Neural Search
```

Comment:
SUper excited to see this page written ;-)   Neural Search is the future.

Reply:
As the author of the documentation obviously I think "Neural Search" as a headline is all right.
I think it's becoming accepted generally by the Information Retrieval community and it's easily recognizable.
I don't have a strong opinion though and I am happy to revise the title if the community prefers.
Let me explain some of my motivations:
1) it's becoming more and more adopted in the industry and in the academy to label search aspects that use directly or byproducts of neural networks(vector embeddings)
2) Dense Vector Search is not wrong per se, but my idea is to extend the documentation in the future with additional neural-based techniques and components, and I see them better documented and reachable if under the same section.
Whenever the section becomes too big (hopefully) we can then split it across many different pages?
3) I don't think the section must be fully comprehensive from the beginning, I think it's all right there are aspects of Neural Search that are not addressed right now. It's actually a good incentive for the community to contribute, improve and extend the support in the future.
4) I agree with this headline we'll get some attention, but I see this is a benefit, a lot of fresh attention to the Apache Solr project can bring more curiosity in and more contributions. I don't think there's anywhere in the documentation that hint at "false advertisement" and that can cause the feeling of under-delivering a larger promise?

Reply:
I put this in Slack, here's my take on it:

I have no objection to calling our dense vector support a neural search feature. To be clear, my objection is naming the entire page neural search when there is only one feature described and no indication whatsoever what other features might be or when they may arrive. Seriously, I didn’t see one thing on that page that’s about anything other than dense vectors.

Of course it’s all right that Solr does not have several neural search features right now. Just as it’s all right we don’t yet have any number of other features. To use a parallel example, LTR is a machine learning feature, but we did not name the page “Machine Learning”; we named the page for exactly the feature. It doesn’t knock against LTR that the page isn’t titled “Machine Learning” nor diminish its significance as a feature.

The Ref Guide is not a marketing document - write a news item or a page in the website to promote this functionality when it is released. Call it “Neural Search Comes to Solr” if you want (and you should!). But the Ref Guide must be clear and direct. It is meant for our users to know *exactly* what they need to do to configure Solr to take advantage of the features. I recently renamed dozens of pages and the guiding principle was to make them named for the exact functionality described on the page to make it clear to readers before they even click the title what they can expect to find there.

SolrCloud and Neural Search can be analogous, as large concepts that have several features. But dense vectors is a single feature. When there is more than one, we can talk about an overarching section that brings all the features together but we aren’t there yet at all.

Reply:
From Slack:
"t makes sense and I you convinced me!
You are the Apache Solr Ref leader for a reason :slightly_smiling_face:
I'll change the documentation page, but I'll keep referring to this (and future) contributions as neural features (for sure it can be considered a "marketing" friendly name, but my genuinely attempt is to use the best reference for this upcoming features, as used around).
I have one final question in relation to a possible next feature(s) arriving.
For example we are planning to contribute an automatic synonyms generation endpoint that leverages neural networks.
At that point in your opinion would be fine to restructure the documentation creating a parent page:
Neural Search with two children?
Dense Vector Retrieval
Neural Network Synonym Generation
ect ect ?
thanks again for the very useful review @Cassandra, much appreciated"